controllers/tektontasks_controller.go

```diff
@@ -175,6 +184,58 @@ func (r *tektonTasksReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	return ctrl.Result{}, nil
 }
 
+func (r *tektonTasksReconciler) checkMultipleCRs(ctx context.Context, tektonRequest *common.Request) (bool, *tekton.TektonTasksList, error) {
+	tektonTasksCRList, err := r.getCRList(ctx)
+	if err != nil {
+		return false, nil, err
+	}
+
+	return len(tektonTasksCRList.Items) > 1, tektonTasksCRList, nil
+}
+
+func (r *tektonTasksReconciler) getCRList(ctx context.Context) (*tekton.TektonTasksList, error) {
+	tektonTasksCRList := &tekton.TektonTasksList{}
+
+	err := r.client.List(ctx, tektonTasksCRList)
+	if err != nil {
+		return nil, err
+	}
+
+	return tektonTasksCRList, nil
+}
+
+func (r *tektonTasksReconciler) setStatusMultipleCRs(ctx context.Context, tektonTasksList *tekton.TektonTasksList) error {
+	errCRs := fmt.Errorf("there are multiple CRs deployed")
```

Comment:
Do we need the error wrapping the message at all or could the message be a const string and be used in the logs and `Message`s?